### PR TITLE
Add commit linter and commitizen

### LINKED ISSUE #38: Add commitizen hook
Enforce commit lint

### PR DESCRIPTION
**Pull request by gtupak**
Closes #38

### CODE REVIEW

File: .husky/pre-commit [hunk 1]
```diff
@@ -0,0 +1,4 @@
+#!/usr/bin/env sh
+. "$(dirname -- "$0")/_/husky.sh"
+
+npm test
```
Comment (gtupak) [inline .husky/pre-commit:4]: Run tests before each commit

File: .husky/commit-msg [hunk 2]
```diff
@@ -0,0 +1,4 @@
+#!/usr/bin/env sh
+. "$(dirname -- "$0")/_/husky.sh"
+
+npx --no -- commitlint --edit "$1"
```
Comment (gtupak) [inline .husky/commit-msg:4]: Actual command that lints the commit message

File: package.json [hunk 3]
```diff
@@ -53,10 +58,17 @@
     "build:esm": "yarn tsc -m es6 --outdir dist/esm",
     "clean": "rm -rf dist",
     "lint": "eslint '{src,test}/**/*.ts'",
-    "test": "nyc mocha 'test/**/*.test.{ts,tsx}'"
+    "test": "nyc mocha 'test/**/*.test.{ts,tsx}'",
+    "prepare": "husky install",
+    "commit": "git add . && cz"
```
Comment (gtupak) [inline package.json:63]: If you want to make a new commit, run `yarn commit` and you'll be walked step by step

File: .husky/pre-commit (hunk 1 above)
Comment (xitronix) [reply .husky/pre-commit:4]: Why don't yarn? And i think it will be annoying. Can we run tests at least before pushing?

File: README.md [hunk 4]
```diff
@@ -1,3 +1,5 @@
+[![Commitizen friendly](https://img.shields.io/badge/commitizen-friendly-brightgreen.svg)](http://commitizen.github.io/cz-cli/)
+
```
Comment (xitronix) [inline README.md:2]: what is it for? :)

**Review by xitronix — DISMISSED** [review]
I played with that a bit. Works for me

File: .husky/pre-commit (hunk 1 above)
Comment (gtupak) [reply .husky/pre-commit]: I agree, I didn't think/know this was an option :)

File: README.md (hunk 4 above)
Comment (gtupak) [reply README.md:2]: To show that this repository is Commitizen friendly. Ie. our commits are clean ;)

**Review by xitronix — APPROVED** [review]

**Pull request merged**